From 63654c934f2b0126267c29bd899f56161b5d41f1 Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Tue, 16 Jul 2013 19:06:34 +0900
Subject: [PATCH 639/675] From: Vladimir Barinov
 <vladimir.barinov@cogentembedded.com>

ML86V7667 always transmits top field first for both PAL and  NTSC -- that makes
application incorrectly  treat interlaced  fields when relying on the standard.
Hence we must set V4L2_FIELD_INTERLACED_TB format explicitly.

Reported-by: Katsuya MATSUBARA <matsu@igel.co.jp>
Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
[Sergei: added a comment.]
Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
---
 drivers/media/video/ml86v7667.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/video/ml86v7667.c b/drivers/media/video/ml86v7667.c
index ebf3b80..9ba2e74 100644
--- a/drivers/media/video/ml86v7667.c
+++ b/drivers/media/video/ml86v7667.c
@@ -209,7 +209,8 @@ static int ml86v7667_try_mbus_fmt(struct v4l2_subdev *sd,
 
 	fmt->code = V4L2_MBUS_FMT_YUYV8_2X8;
 	fmt->colorspace = V4L2_COLORSPACE_SMPTE170M;
-	fmt->field = V4L2_FIELD_INTERLACED;
+	/* The top field is always transferred first by the chip */
+	fmt->field = V4L2_FIELD_INTERLACED_TB;
 	fmt->width = 720;
 	fmt->height = priv->std & V4L2_STD_525_60 ? 480 : 576;
 
-- 
1.8.2

